### PR TITLE
Port standard-syntax-table.

### PR DESCRIPTION
**Pull request by jaccarmac**
Address #969.

I'm watching #985 with great interest as well.

### CODE REVIEW

**Comment by jaccarmac** [comment]
Trying to get back into the swing of things after not contributing in forever. Does the hubbub around Vstandard_syntax_table have anything to do with the new variable stuff?

File: rust_src/src/syntax.rs [hunk 1]
```diff
@@ -59,4 +59,11 @@ fn check_syntax_table_p(table: LispCharTableRef) {
     }
 }
 
+/// Return the standard syntax table.  This is the one used for new buffers.
+#[lisp_fn]
+pub fn standard_syntax_table() -> LispCharTableRef {
+    let buffer_table = unsafe { buffer_defaults.syntax_table_ };
```
Comment (jaccarmac) [inline rust_src/src/syntax.rs]: I don't particularly like this but it seems to be the only good way to do this. Also prefer the type "conversion" over returning raw LispObject, but I see in other places with direct returns of variables of known types Rust just sees a LispObject. I suppose this is a problem that will slowly become less of one as more C turns into Rust?

**Comment by shaleh** [comment]
> Trying to get back into the swing of things after not contributing in forever. Does the hubbub around Vstandard_syntax_table have anything to do with the new variable stuff?

No. In the C it is a macro to read a member of struct buffer. It was ported as an object incorrectly.

File: rust_src/src/syntax.rs (hunk 1 above)
Comment (shaleh) [reply rust_src/src/syntax.rs]: As I mentioned on brotzeit's PR, perhaps representing `Vstandard_syntax_table` as `standard_syntax_table!` is a reasonable choice.

Comment (shaleh) [reply rust_src/src/syntax.rs]: Simplified:

```
/// Return the standard syntax table.  This is the one used for new buffers.
#[lisp_fn]
pub fn standard_syntax_table() -> LispCharTableRef {
    unsafe { buffer_defaults.syntax_table_ }.into()
}
```

Comment (jaccarmac) [reply rust_src/src/syntax.rs]: Ooh nice, I'll change that. Thanks!

Re: the macro. In that case we would need a macro for every member of buffer_defaults, as with C. Would it be worthwhile to create a wrapper for buffer_defaults without underscore tails? Creating a full-on equivalent to the C macro system is possible of course, just not sure exactly what it gets us.

(I'm not familiar with the Vx variables in Emacs, maybe there's some uniformity between them i am missing?)

Comment (shaleh) [reply rust_src/src/syntax.rs]: I have not researched the C code enough to have an opinion yet on this.

In general, I like having easily greppable names. But there is a cost v. benefit analysis for sure.
`buffer_defaults.something...` is not as obvious as `standard_something`.

Comment (jaccarmac) [reply rust_src/src/syntax.rs]: Fair enough, and easy enough to add to `remacs_sys` or somesuch. What do you think is the best way to deal with `unsafe` in that case? Leave the caller to mark `unsafe` as appropriate?

Comment (shaleh) [reply rust_src/src/syntax.rs]: Usually best.

Also, remas-sys twiddling should not be required for most things any more. If you need something from C make sure it is exported from a C header. bindgen will update the Rust side for you on the next build.

Comment (shaleh) [reply rust_src/src/syntax.rs]: There is no need for `BVAR` in the current code. Turn the LispObject into a LispBufferRef and then directly access the field. bindgen has setup the C structs for us.

Comment (shaleh) [reply rust_src/src/syntax.rs]: The advantage of `Vstandard_syntax_table` is it abstracts away the buffer and the C struct member name. So it is obviously "getting the standard syntax table" and not "um, accessing a field on a buffer. What does that field do....."

**Pull request merged**